cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -1441,6 +1452,14 @@ Result<FileInfo> GetContainerPropsAsFileInfo(const AzureLocation& location,
       info.set_type(FileType::NotFound);
       return info;
     }
+
+    if (IsForbidden(exception)) {
+      // User delegated sas tokens don't allow for getting container properties. Assume
+      // the container exists
+      info.set_type(FileType::Directory);
+      info.set_mtime(std::chrono::system_clock::now());
+      return info;
+    }
```

Comment:
What is happen when we access to nonexistent container? Is not forbidden response is returned?

Reply:
Oh...